src/connector/src/connector_common/connection.rs

```diff
@@ -250,7 +250,13 @@ impl Connection for IcebergConnection {
             }
         }
 
-        // test catalog
+        // If there is no catalog type, we won't test the catalog.
+        // The iceberg connection could be used to provide warehouse information only.
```

Comment:
According to today's talk, we won't deliver a stage with self-hosted JDBC catalog; as a result, I think the iceberg connection must provide both warehouse information and catalog info.

Reply:
Correct.